clang/lib/DPCT/Utility.h

```diff
@@ -587,6 +589,20 @@ class PairedPrinter {
   ~PairedPrinter() { OS << Postfix; }
 };
 std::string appendPath(const std::string &P1, const std::string &P2);
+class CheckedOfstream : public std::ofstream {
+  std::string FileName;
+
+public:
+  CheckedOfstream(const std::string &FileName,
+                  ios_base::openmode Mode = ios_base::out)
+      : std::ofstream(FileName, Mode), FileName(FileName) {}
+
+  ~CheckedOfstream();
+};
+bool writeDataToFile(const std::string &FileName, const std::string &Data);
+bool appendDataToFile(const std::string &FileName, const std::string &Data);
+std::error_code createDirectories(const clang::tooling::UnifiedPath &FilePath,
+                                  bool IgnoreExisting = true);
```

Comment:
I prefer that these 3 functions return void, because if there is error program will exit. Since only successful case these functions return, the return value is unnecessary.
Others LGTM.